### PR TITLE
create tab

### PR DESCRIPTION
**Pull request by jdorweiler**
**Reviewer:**

## Description:

Currently we're only creating new tabs when a request comes through. This can be a problem when you try to view the popup before a request has been made, like on the new-tab page. 

I updated the tabs.onUpdated listener to create a new tab object if one doesn't exist yet. 

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: js/background.js [hunk 1]
```diff
@@ -145,10 +145,14 @@ chrome.webRequest.onBeforeRequest.addListener(
...
-      
+
+      // skip requests to background tabs
+      if(e.tabId === -1){
+          return;
+      }
+
       if(!tabs[e.tabId]){
-          tabs[e.tabId] = {'trackers': {}, "total": 0, 'url': e.url, "dispTotal": 0}
-          updateBadge(e.tabId, tabs[e.tabId].dispTotal);
```
Comment (russellholt) [inline js/background.js]: So you won't update the badge with an initial 0 anymore?

File: js/background.js [hunk 2]
```diff
@@ -200,10 +204,12 @@ function updateBadge(tabId, numBlocked){
...
 
 chrome.tabs.onUpdated.addListener(function(id, info, tab) {
+    // an existing tab has been reloaded or changed url, clear tab data and set status
     if(tabs[id] && info.status === "loading" && tabs[id].status !== "loading"){
         tabs[id] = {'trackers': {}, "total": 0, "dispTotal": 0, 'url': tab.url, "status": "loading"};
         updateBadge(id, 0);
     }
+    // existing tab, update the status and url
     else if(tabs[id] && info.status === "complete"){
         tabs[id].status = "complete";
```
Comment (russellholt) [inline js/background.js]: No updating the badge here?

File: js/background.js (hunk 1 above)
Comment (jdorweiler) [inline js/background.js]: I don't think we need to do it here but it doesn't hurt so I'll add it back.

File: js/background.js (hunk 2 above)
Comment (jdorweiler) [inline js/background.js]: The badge is set to zero in the `if` above and below when the tab is created. After that it's updated when anything new is blocked in the onBeforeRequest listener.

**Comment by russellholt** [comment]
tested.
all cases check out!

**Pull request merged**